server/bastionlab_polars/src/lib.rs

```diff
@@ -76,13 +77,17 @@ impl DataFrameArtifact {
 pub struct BastionLabPolars {
     dataframes: Arc<RwLock<HashMap<String, DataFrameArtifact>>>,
     sess_manager: Arc<SessionManager>,
+    persist_tracking: Arc<RwLock<HashMap<String, Vec<Instant>>>>,
```

Comment:
Maybe refactor this to use a single data structure instead of 2?

Reply:
Also, I can't really see why these checks cannot be done in the SessionManager instead of here

Reply:
I use two structures because I want to track DoS attempts on storage and memory separately